Analysis/Smells/AstSmells/LongScopeChaining.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Esprima;
+using Esprima.Ast;
+
+namespace BSDetector.Analysis.Smells.AstSmells
+{
+    public class LongScopeChaining : AstSmell
+    {
+        public override string SmellName => "LONG_SCOPE_CHAINING";
+
+        public override void AnalyzeNode(INode node, int levelOfFunction)
+        {
+            if (node is FunctionDeclaration FunctionDeclarationNode)
+            {
+                levelOfFunction += 1;
+                foreach (var child in node.ChildNodes)
+                    AnalyzeNode(child, levelOfFunction, FunctionDeclarationNode.Location);
+            }
```

Comment:
I'm not sure, but you can try to reduce lines of code by using ++levelOfFunction as a parameter for AnalyzeNode.